addon/waiter-manager.ts

```diff
@@ -0,0 +1,79 @@
+import { IWaiter, IPendingWaiterState } from './types';
+
+const WAITERS = new Set();
+
+/**
+ * Registers a waiter with the test waiter manager.
+ *
+ * @public
+ * @param waiter {IWaiter} A test waiter instance
+ */
+export function register(waiter: IWaiter): void {
+  WAITERS.add(waiter);
+}
+
+/**
+ * Unregisters a waiter with the test waiter manager.
+ *
+ * @public
+ * @param waiter {IWaiter} A test waiter instance
+ */
+export function unregister(waiter: IWaiter): void {
+  WAITERS.delete(waiter);
+}
+
+/**
+ * Gets an array of all test waiters current registered.
+ *
+ * @public
+ * @returns {IWaiter[]}
+ */
+export function getWaiters(): IWaiter[] {
+  return [...WAITERS];
+}
+
+/**
+ * Clears all test waiters from the waiter manager.
+ *
+ * @public
+ */
+export function reset(): void {
+  WAITERS.clear();
+}
+
+/**
+ * Gets the current state of all test waiters. Any test waiters whose
+ * `waitUntil` method returns false will be considered `pending`.
+ *
+ * @returns a result object containing a count of all waiters pending and
```

Comment:
Is there a better way to type this (for docs I mean)? Can we give the return type an interface, and reference it here?

Reply:
Done.